services/workflows-service/src/secrets-manager/aws-secrets-manager.ts

```diff
@@ -0,0 +1,107 @@
+import { SecretsManager } from '@/secrets-manager/secrets-manager';
+import {
+  CreateSecretCommand,
+  GetSecretValueCommand,
+  PutSecretValueCommand,
+  ResourceNotFoundException,
+  SecretsManagerClient,
+} from '@aws-sdk/client-secrets-manager';
+import { z } from 'zod';
+
+const SecretStringSchema = z
+  .string()
+  .transform((value, ctx) => {
+    try {
+      return JSON.parse(value);
+    } catch (error) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: 'Secret value is not a valid JSON',
+      });
+
+      return z.NEVER;
+    }
+  })
+  .pipe(z.record(z.string(), z.string()));
+
+export class AwsSecretsManager implements SecretsManager {
+  private client: SecretsManagerClient;
+  private prefix: string;
+  private customerId: string;
+
+  constructor({
+    prefix,
+    customerId,
+    awsRegion,
+    awsAccessKeyId,
+    awsSecretAccessKey,
+  }: {
+    prefix: string;
+    customerId: string;
+    awsRegion: string;
+    awsAccessKeyId: string;
+    awsSecretAccessKey: string;
+  }) {
+    this.client = new SecretsManagerClient({
+      region: awsRegion,
+      credentials: {
+        accessKeyId: awsAccessKeyId,
+        secretAccessKey: awsSecretAccessKey,
+      },
+    });
```

Comment:
I think we can remove this access key and secret for aws ... Their sdk should consume the credentials automaticlly 

@MayurDuduka Please correct us if I wrong